### PR TITLE
Feature: Action Plan Bar

### PR DESCRIPTION
**Pull request by rachelchenrc**
[Trello card](https://trello.com/c/enZw6MbQ/253-as-a-user-i-want-to-see-my-progress-via-the-action-plan-on-the-dashboard)
[Live env](https://nj-career-network-dev4.firebaseapp.com/dashboard/)

- store action plan (thresholds) in user as actionPlan

- Add weekly counts to the user's stats
weeklyActivitiesCount
weeklyApplicationsCount
weeklyTasksCount

- Reset the action plan to default values and weekly counts to 0 on Monday 00:00:00 AM

Screen Shot 2020-05-26 at 8 47 12 AM

### CODE REVIEW

File: components/dashboard/ActionPlan/ActionPlanBar.js [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+    textAlign: 'left',
+    marginLeft: theme.spacing(1),
+  },
+}));
+
+function ActionPlanBar({ userStats, actionPlan }) {
+  const classes = useStyles();
+  const { userDocRef } = useAuth();
+
+  function startOfWeek() {
```
Comment (hannahtwobulls) [inline components/dashboard/ActionPlan/ActionPlanBar.js:60]: The `date-fns` library actually has a `startOfWeek()` function that might be worth looking at! [HERE](https://date-fns.org/v2.13.0/docs/startOfWeek)

File: components/activityInput/ActivityInputDialog.js [hunk 2]
```diff
@@ -238,6 +238,8 @@ function ActivityInputDialog({ fullScreen, show, onClose }) {
       const stats = {
         activityLogEntriesCount: firebase.firestore.FieldValue.increment(1),
         activityLogEntriesLatestTimestamp: firebase.firestore.FieldValue.serverTimestamp(),
+        weeklyActivitiesCount: firebase.firestore.FieldValue.increment(1),
```
Comment (hannahtwobulls) [inline components/activityInput/ActivityInputDialog.js:241]: Are the `weeklySomethingCount` what Gideon had as `current` in his POC?

Comment (rachelchenrc) [reply components/activityInput/ActivityInputDialog.js:241]: yes, serve the same purpose.

File: components/dashboard/ActionPlan/ActionPlanBar.js (hunk 1 above)
Comment (rachelchenrc) [reply components/dashboard/ActionPlan/ActionPlanBar.js]: nice, use startOfWeek() from date-fns instead.

File: components/activityInput/ActivityInputDialog.js (hunk 2 above)
Comment (hannahtwobulls) [reply components/activityInput/ActivityInputDialog.js:241]: We might want to keep the naming the same just for clarity and to make comparisons easier (ex: checking the `actionPlan.current.goals` with `actionPlan.target.goals`. Let me know your thoughts

**Pull request merged**